udata_metrics/tasks.py

```diff
@@ -10,79 +12,88 @@
     '''
     paginate on target endpoint
     '''
+    url = f'{current_app.config["METRICS_API"]}/{target}_total/data/'
+    url += f'?{value_key}__greater=1&page_size={page_size}'
+
     with requests.Session() as session:
-        url = f'{current_app.config["METRICS_API"]}/{target}_total/data/'
-        url += f'?{value_key}__greater=1&page_size=50'
-        r = session.get(url, timeout=10)
-        r.raise_for_status()
-        data = r.json()
-        log.info(f'{data["meta"]["total"]} objects found')
-        for row in data['data']:
-            yield row
-        while data['links'].get('next'):
-            r = session.get(data['links'].get('next'), timeout=15)
+        while url is not None:
+            r = session.get(url, timeout=10)
             r.raise_for_status()
             data = r.json()
+            
             for row in data['data']:
                 yield row
 
+            url = data['links'].get('next')
 
-def process_metrics_result(target_endpoint: str,
-                           model: db.Document,
-                           id_key: str,
-                           value_key: str = 'visit') -> None:
-    '''
-    Fetch metrics and update udata objects with the total metrics count
-    '''
-    log.info(f'Processing model {model}')
-    start = datetime.now()
-    for data in iterate_on_metrics(target_endpoint, value_key):
-        if model.__name__ == 'Resource':
-            # Specific case for resource:
-            # - it could either be a Dataset Resource embedded document or a CommunityResource
-            # - it requires special performance improvement to prevent saving the entire document
-            modified_count = Dataset.objects(resources__id=data[id_key]).update(
-                **{'set__resources__$__metrics__views': data[value_key]}
-            )
-            if not modified_count:
-                # No embedded resource found with this id, could be a CommunityResource
-                save_model(CommunityResource, model_id=data[id_key], value=data[value_key])
+@log_timing
+def update_resources_and_community_resources():
+    sum_of_resources_downloads = {}
+
+    for data in iterate_on_metrics("resources", "download_resource"):
+        if data['dataset_id'] is None:
+            save_model(CommunityResource, data['resource_id'], 'views', data['download_resource'])
         else:
-            save_model(model, model_id=data[id_key], value=data[value_key])
-    log.info(f'Done in {datetime.now() - start}')
+            sum_of_resources_downloads.setdefault(data['dataset_id'], 0)
+            sum_of_resources_downloads[data['dataset_id']] += data['download_resource']
+
+            Dataset.objects(resources__id=data['resource_id']).update(
+                **{f'set__resources__$__metrics__views': data['download_resource']}
+            )
+
+    for dataset_id, sum in sum_of_resources_downloads.items():
+        save_model(Dataset, dataset_id, 'number_of_resources_downloads', sum)
```

Comment:
Unsure about the wording, but would remove the `number of`
```suggestion
        save_model(Dataset, dataset_id, 'resources_downloads', sum)
```

Reply:
Additionally, I think we should add it to [metrics keys defined in dataset model](https://github.com/opendatateam/udata/blob/0ddcfb47e2d94b96da35c1c37239ce3cac109006/udata/core/dataset/models.py#L503)?

Reply:
Done in 0fcba86

Reply:
I think my second comment is still valid?

Reply:
Yes but it's inside `udata`? I need to do a PR there too

Reply:
Done in https://github.com/opendatateam/udata/pull/3042